buffer/src/main/java/io/netty/buffer/PoolArena.java

```diff
@@ -618,10 +626,19 @@ private void destroyPoolChunkLists(PoolChunkList<T>... chunkLists) {
 
     static final class HeapArena extends PoolArena<byte[]> {
         private final AtomicReference<PoolChunk<byte[]>> lastDestroyedChunk;
+        private final Queue<PooledByteBuf<byte[]>> bufferQueue;
+        private final ObjectPool.Handle<PooledHeapByteBuf> handle;
 
         HeapArena(PooledByteBufAllocator parent, SizeClasses sizeClass) {
             super(parent, sizeClass);
             lastDestroyedChunk = new AtomicReference<PoolChunk<byte[]>>();
+            bufferQueue = PlatformDependent.newFixedMpmcQueue(ARENA_BUFFER_QUEUE_CAPACITY);
```

Comment:
The mpmc queues are not really "free" (they have lot of padding). So, few comments:
- try to allocate them lazily or this will cost (on the heap) a lot even when not used
- try using the unpadded variants (which I have implemented in JCTools - so should be somewhere available..)
- no need of mpmc but mpsc should be enough - and check if we expect `newByteBuf` to be somehow protected by some lock (it should AFAIK!)

Reply:
> no need of mpmc but mpsc should be enough - and check if we expect newByteBuf to be somehow protected by some lock (it should AFAIK!)

The 4.1-branch use `RECYCLER.get()` to create ByteBuf in `newByteBuf()`, so should not need a lock, no?

Reply:
>  The 4.1-branch use RECYCLER.get() to create ByteBuf in newByteBuf(), so should not need a lock, no

I mean in the Arena's constructor: you still create a queue + handle which won't be used if no virtual threads are employed, if I read this right

Reply:
> I mean in the Arena's constructor: you still create a queue + handle which won't be used if no virtual threads are employed, if I read this right

Because this PR is checking the thread type(whether virtual or not) at runtime, did you mean when the first virtual thread be detected to call `newByteBuf()`, then create a `queue` and a `handle`?

Reply:
>But there can be add a 'JAVA_VERSION' check in the constructor

It's not enough because is not fair to assume that every user with JDK 19 is going to use virtual threads either.

Reply:
yep - we need, for JDK >=19 be able to allocate the queue/handle at runtime, since arenas are not "owned" by virtual threads but shared - unless we use a single "always" the same mechanism as efficient as thread local

Reply:
Arenas could be bound to v threads: but not necessarly that's why always allocating the queue could be wastefull - since it depends if users, despite been able to use virtual threads, maybe they won't use them

Reply:
Got it, then we may add a system property to let the user decide whether enable virtual thread support? e.g. "io.netty.allocator.supportVirtualThread" ?